src/main/java/io/ebeaninternal/server/expression/InExpression.java

```diff
@@ -100,15 +110,32 @@ public void addSql(SpiExpressionRequest request) {
       prop = null;
     }
 
+    String realPropName = propName;
+    if (containsNull != not) {
+      request.append("(");
+    }
     if (prop != null) {
-      request.append(prop.getAssocIdInExpr(propName));
+      realPropName = prop.getAssocIdInExpr(propName);
+      request.append(realPropName);
       String inClause = prop.getAssocIdInValueExpr(not, bindValues.length);
       request.append(inClause);
 
     } else {
       request.append(propName);
       request.appendInExpression(not, bindValues);
     }
+    // if we perform an "in" query (not = false) and we want "null's",
+    // we must append an additional OR statement.
+    // if we perform a "not in" query (not = true) and we have not nulls
+    // in our values list, we must include nulls.
+    // Consider, the db has the values 1,2,3,4,null:
+    // in(1,2)          => 1,2
+    // in(1,2,null)     => 1,2,null  (normal SQL would return 1,2 only)
+    // not in(1,2)      => 3,4,null  (normal SQL would return 3,4 only)
+    // not in(1,2,null) => 3,4       (normal SQL would return 3,4 only)
```

Comment:
I hope this will not break too much...
my argument is, that `.in(null)` / `notIn(null)` should be equivalent to `.eq(null)` / `.ne(null)`
For eq/ne there is already code, that handles the null case